File: jbake-core/src/test/java/org/jbake/util/ConfigurationPrinterTest.java
package org.jbake.util;

import org.jbake.TestUtils;
import org.jbake.app.configuration.JBakeConfiguration;
import org.jbake.app.configuration.JBakeConfigurationFactory;
import org.junit.jupiter.api.Test;

import java.io.ByteArrayOutputStream;
import java.io.PrintStream;

import static org.assertj.core.api.Assertions.assertThat;

class ConfigurationPrinterTest {

    @Test
    void shouldPrintHeader() throws Exception {
        JBakeConfiguration configuration = new JBakeConfigurationFactory().getConfigUtil().loadConfig(TestUtils.getTestResourcesAsSourceFolder());
        ByteArrayOutputStream data = new ByteArrayOutputStream();
        PrintStream out = new PrintStream(data);
        ConfigurationPrinter printer = new ConfigurationPrinter(configuration, out);

        printer.print();

        assertThat(data.toString()).contains("DEFAULT - Settings");
        assertThat(data.toString()).contains("CUSTOM - Settings");
        assertThat(data.toString()).contains("Key");
        assertThat(data.toString()).contains("Value");
    }


    @Test
    void shouldPrintKeyAndValue() throws Exception {
        JBakeConfiguration configuration = new JBakeConfigurationFactory().getConfigUtil().loadConfig(TestUtils.getTestResourcesAsSourceFolder());
        ByteArrayOutputStream data = new ByteArrayOutputStream();
        PrintStream out = new PrintStream(data);
        ConfigurationPrinter printer = new ConfigurationPrinter(configuration, out);

        printer.print();

        assertThat(data.toString()).contains("site.host");
        assertThat(data.toString()).contains("http://www.jbake.org");
    }
}

File: jbake-core/src/test/java/org/jbake/util/ConfigurationPrinterTest.kt
package org.jbake.util

import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.string.shouldContain
import org.jbake.TestUtils
import org.jbake.app.configuration.JBakeConfigurationFactory
import java.io.ByteArrayOutputStream
import java.io.PrintStream

internal class ConfigurationPrinterTest : StringSpec({

    "shouldPrintHeader" {
        val configuration =
            JBakeConfigurationFactory().configUtil.loadConfig(TestUtils.testResourcesAsSourceDir)
        val data = ByteArrayOutputStream()
        val out = PrintStream(data)
        val printer = ConfigurationPrinter(configuration, out)

        printer.print()

        val output = data.toString()
        output shouldContain "DEFAULT - Settings"
        output shouldContain "CUSTOM - Settings"
        output shouldContain "Key"
        output shouldContain "Value"
    }


    "shouldPrintKeyAndValue" {
        val configuration =
            JBakeConfigurationFactory().configUtil.loadConfig(TestUtils.testResourcesAsSourceDir)
        val data = ByteArrayOutputStream()
        val out = PrintStream(data)
        val printer = ConfigurationPrinter(configuration, out)

        printer.print()

        val output = data.toString()
        output shouldContain "site.host"
        output shouldContain "http://www.jbake.org"
    }
})
